python/tvm/relax/frontend/onnx_frontend.py

```diff
@@ -0,0 +1,1150 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, import-self, len-as-condition, unused-argument, too-many-lines
+# pylint: disable=import-outside-toplevel
+"""ONNX: Open Neural Network Exchange frontend for Relax."""
+import math
+import warnings
+from typing import Union, Optional
+
+import numpy as _np
+
+import tvm
+from tvm import relax, topi, relay
+from tvm.target import Target
+from tvm.ir import IRModule
+from tvm.relax import testing, PyExprMutator
+from tvm._ffi import base as _base
+from tvm.runtime import ndarray as _nd
+from tvm.relay.expr import TupleWrapper, Var, GlobalVar
+from tvm.relay.frontend.onnx import OnnxOpConverter as RelayOnnxOpConverter
+
+
+def new_var(var_name, shape, dtype="float32"):
+    return testing.nn.Parameter(shape=shape, dtype=dtype, name=var_name)
+
+
+def get_type(elem_type):
+    """Converts onnx integer datatype to numpy datatype"""
+    # If a string was passed instead of a tensor type, it does not need
+    # conversion and can be returned.
+    if isinstance(elem_type, str):
+        return elem_type
+
+    try:
+        from onnx.mapping import TENSOR_TYPE_TO_NP_TYPE
+    except ImportError as e:
+        raise ImportError("Unable to import onnx which is required {}".format(e))
+
+    try:
+        from onnx import TensorProto
+    except ImportError as e:
+        raise ImportError("Unable to import TensorProto from onnx {}".format(e))
+
+    return str(TENSOR_TYPE_TO_NP_TYPE[elem_type])
+
+
+def get_info(info_proto):
+    """Extract the shape from a ValueInfoProto."""
+    shape = []
+    shape_name = []
+    for dim in info_proto.type.tensor_type.shape.dim:
+        name = dim.dim_param
+        value = dim.dim_value
+        if value is None or value == 0:
+            value = tvm.tir.Var("d", "int64")
```

Comment:
why name the var "d"?

Reply:
I think we usually use 'd' for dynamic, although I'm not sure..

Reply:
Yeah its just a stand-in dynamic shape. I've updated it to be `dyn` for more clarity.